test/e2e/router_status_test.go

```diff
@@ -0,0 +1,297 @@
+	// Warning: Deleting ingress so quickly will cause race conditions in updating ingress status and
+	//          cause some errors to be thrown; however, it doesn't matter the purpose of this E2E test.
+	assertIngressControllerDeleted(t, kclient, ic)
+
+	// Wait until route status goes unadmitted.
+	if err := waitForRouteStatusClear(t, kclient, routeName, ic.Name); err != nil {
+		t.Fatalf("failed to observe route status clear when ingress controller was deleted: %v", err)
+	}
+}
+
+// TestIngressControllerRouteSelectorUpdateShouldClearRouteStatus modifies an ingress controller's selectors to
+// unadmit a route and verifies it clears the admitted route's status.
+func TestIngressControllerRouteSelectorUpdateShouldClearRouteStatus(t *testing.T) {
+	t.Parallel()
+	// Create an ingress controller that can admit our route.
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "ic-route-selector-test"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
+	ic.Spec.RouteSelector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"type": "foo",
+		},
+	}
+
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	// Create a route to be immediately admitted by this ingress controller and then when the IC label selectors are
+	// updated, the status should clear.
+	// Use openshift-console namespace to get a namespace outside the ingress-operator's cache.
+	routeFooLabelName := types.NamespacedName{Namespace: "openshift-console", Name: "route-foo-label"}
+	routeFooLabel := newRouteWithLabel(routeFooLabelName, "foo")
+	if err := kclient.Create(context.TODO(), routeFooLabel); err != nil {
+		t.Fatalf("failed to create route: %v", err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), routeFooLabel); err != nil {
+			t.Fatalf("failed to delete route %s: %v", routeFooLabelName, err)
+		}
+	}()
+
+	// Create a route that will NOT be immediately admitted by the ingress controller, but will be admitted AFTER
+	// the IC selectors are updated. The status SHOULD be successfully admitted.
+	// Use openshift-console namespace to get a namespace outside the ingress-operator's cache.
+	routeBarLabelName := types.NamespacedName{Namespace: "openshift-console", Name: "route-bar-label"}
+	routeBarLabel := newRouteWithLabel(routeBarLabelName, "bar")
+	if err := kclient.Create(context.TODO(), routeBarLabel); err != nil {
+		t.Fatalf("failed to create route: %v", err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), routeBarLabel); err != nil {
+			t.Fatalf("failed to delete route %s: %v", routeBarLabel, err)
+		}
+	}()
+
+	// Wait for routeFooLabel to be admitted upon creation.
+	admittedCondition := routev1.RouteIngressCondition{Type: routev1.RouteAdmitted, Status: corev1.ConditionTrue}
```

Comment:
You could declare a `routeAdmittedStatusCondition` variable at the top of the file to avoid the need to declare the same variable in all three tests.